File: py/IEC61968/Common/Person.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106
from CIM_STD_PYTHON.TC57CIM.IEC61968.Common.TelephoneNumber import TelephoneNumber
from CIM_STD_PYTHON.TC57CIM.IEC61968.Common.ElectronicAddress import ElectronicAddress
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Core.IdentifiedObject import IdentifiedObject

class Person(IdentifiedObject):
    
    def __init__(self):
        super().__init__()
        self.electronic_address = ElectronicAddress()
        self.first_name = ""
        self.landline_phone = TelephoneNumber()
        self.last_name = ""
        self.middle_name = ""
        self.mobile_phone = TelephoneNumber()
        self.prefix = ""
        self.special_need = ""
        self.suffix = ""
        self.roles = ""

File: java/IEC61968/Common/Person.java
package IEC61968.Common;

import IEC61970.Base.Domain.String;
import IEC61970.Base.Core.IdentifiedObject;

/**
 * General purpose information for name and other information to contact people.
 * @author T. Kostic
 * @version 1.0
 * @created 25-Dec-2023 8:45:23 PM
 */
public class Person extends IdentifiedObject {

	/**
	 * Electronic address.
	 */
	public ElectronicAddress electronicAddress;
	/**
	 * Person's first name.
	 */
	public String firstName;
	/**
	 * Landline phone number.
	 */
	public TelephoneNumber landlinePhone;
	/**
	 * Person's last (family, sir) name.
	 */
	public String lastName;
	/**
	 * Middle name(s) or initial(s).
	 */
	public String mName;
	/**
	 * Mobile phone number.
	 */
	public TelephoneNumber mobilePhone;
	/**
	 * A prefix or title for the person's name, such as Miss, Mister, Doctor, etc.
	 */
	public String prefix;
	/**
	 * Special service needs for the person (contact) are described; examples include
	 * life support, etc.
	 */
	public String specialNeed;
	/**
	 * A suffix for the person's name, such as II, III, etc.
	 */
	public String suffix;
	/**
	 * All roles of this person.
	 */
	public PersonRole Roles;

	public Person(){

	}
}//end Person